duckdb-split v1.2.0

flaviomartins opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->



This PR drops the extensions fts and https conda-forge packages. The automatic update failed because these extensions moved out-of-tree.

I will also try to build the parquet extension built-in. Our build scripts mention it is built-in by default but I don't think it is being linked.

Closes #8, #9, #19

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

flaviomartins commented:
Upstream releases DONT_LINK tpcds, tpch. See: https://github.com/duckdb/duckdb/blob/v1.2-histrionicus/.github/config/bundled_extensions.cmake

I tried to match the split of this feedstock using the mechanism upstream uses configure linked vs not-linked extensions. Should we match upstream and link icu, json, and autocomplete and drop the separate packages?

flaviomartins commented:
Comparing the size of the linux_64 libduckdb artifact I see a 5MB increase. I believe this is mostly from the icu extension. Even though upstream releases this as a built in extension we might want to keep it separate in a different package.

moritzwilksch commented:
thanks for addressing these issues! Is the plan to maintain the `httpfs` extension in a separate feedstock then?

flaviomartins commented:
> thanks for addressing these issues! Is the plan to maintain the `httpfs` extension in a separate feedstock then?

I am looking for feedback on this. As it is, we "simply" stop building/shipping the extensions for `httpfs` and `fts`. As with all other extensions, these would now need to be installed using duckdb's own extension management: 

   I enabled the following options to help the transition:

   ```
   -DENABLE_EXTENSION_AUTOLOADING=1 \
   -DENABLE_EXTENSION_AUTOINSTALL=1 \
   ```

We could decide we want to build/ship these extensions ourselves (in conda-forge). In this case, we need to create duckdb-httpfs-feedstock and duckdb-fts-feedstock
   
See:
   https://github.com/duckdb/duckdb-httpfs
   https://github.com/duckdb/duckdb-fts

moritzwilksch commented:
I think we should build them in separate feedstocks then.
In enterprise environments, fetching compiled extensions from a DuckDB server on the internet might not be possible. Shipping them on conda-forge is super valuable if an internal conda mirror is available. I’d usually be happy to help, but I don’t know enough about the build process to contribute much 😅
Since the extensions have moved to a separate repo upstream I also thinks it's fair to have them in a separate feedstock

flaviomartins commented:
Ready for review

flaviomartins commented:
@conda-forge-admin, please rerender

flaviomartins commented:
@janjagusch @xhochy Have you had the chance to take a look at this? I’d love your feedback.

flaviomartins commented:
@conda-forge-admin, please rerender

janjagusch commented on recipe/meta.yaml:
```diff
@@ -80,125 +80,9 @@ outputs:
     test:
       commands:
         - duckdb -version
-  - name: duckdb-extension-json
```
In your PR description you mention dropping the fts and https extensions.

> This PR drops the extensions fts and https conda-forge packages. The automatic update failed because these extensions moved out-of-tree.

But the diff suggests that you drop a lot more extensions than that. What is your plan with these extensions?

xhochy replied:
JSON is now a built-in extension and will no longer be separated installed.

janjagusch commented on recipe/build.sh:
```diff
@@ -27,19 +27,32 @@ fi
-    -DSKIP_EXTENSIONS="${SKIP_EXTENSIONS}" \
-    -DDISABLE_BUILTIN_EXTENSIONS=ON \
+    -DDUCKDB_EXTENSION_CONFIGS="$PWD/bundled_extensions.cmake" \
+    -DENABLE_EXTENSION_AUTOLOADING=1 \
+    -DENABLE_EXTENSION_AUTOINSTALL=1 \
```
We should not set these flags. The way I understand it, this will make DuckDB 'phone home' to its extension registry, which might not be possible from locked-down corporate environments.

janjagusch reviewed (commented):
Thanks for your PR, @flaviomartins. I just got back from vacation and finally had some time to give this a brief review.

> We could decide we want to build/ship these extensions ourselves (in conda-forge). In this case, we need to create duckdb-httpfs-feedstock and duckdb-fts-feedstock

Why not ship the extensions as part of this feedstock? What's the advantage of creating separate feedstocks?

xhochy commented:
I would include the https and ftpfs in this feedstock if they are released as part of the main duckdb release. If they will diverge in their release cycles, we can separate them out but from experience of other complex feedstocks, separation is often making things even harder.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ The outputs section contained an unexpected subsection name. noarch is not a valid subsection name.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14174120963](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14174120963). Examine the logs at this URL for more detail.</sub>

xhochy commented on recipe/test-builtin-extensions.sh:
```diff
@@ -3,9 +3,10 @@ set -euxo pipefail
 
-QUERY="select extension_name from duckdb_extensions() where (installed and install_mode = 'STATICALLY_LINKED');"
+# jemalloc is statically linked into DuckDB only on the Linux platform. Skip it here for simplicity.
+QUERY="select extension_name from duckdb_extensions() where (installed and install_mode = 'STATICALLY_LINKED' and extension_name <> 'jemalloc');"
```
We should make the test depend on the OS instead.

flaviomartins replied:
I am not sure how to do it right

flaviomartins commented:
The README.md mentions all the extensions still

xhochy commented:
I want to add them again before we merge. But let's first get the current state green.

xhochy commented:
TODOs before merge:

- [ ] Get the JSON extension green again here
- [ ] Build `icu` with `WITH_INTERNAL_ICU=OFF` to ensure that the binary stays small
- [ ] Build `httpfs` and `fts` extensions again.

xhochy commented:
`WITH_INTERNAL_ICU` is only going to be in effect with the next release. If this passes, we can merge.
